tests/meta/demultiplex/conftest.py

```diff
@@ -401,6 +408,22 @@ def fixture_flow_cell_directory_name_demultiplexed_with_bcl_convert_flat(
     return f"230505_A00689_0804_B{flow_cell_name_demultiplexed_with_bcl_convert}"
 
 
+@pytest.fixture(
+    name="flow_cell_directory_name_demultiplexed_with_bcl_convert_on_sequencer", scope="session"

```

Comment:
This is not how pytest fixtures are commonly used (using the fixture prefix and renaming them with the name parameter).

There is also no need to change the scope of the fixture from the default to session unless it is expensive to setup.

Reply:
This might be an opportunity for us to change how we write our fixtures. What do you think @Vince-janv, @diitaz93, @henrikstranneheim, @islean, @ChrOertlin?

Reply:
I'm in favour of skipping fixture in the function name, and also skipping the `name` parameter. I don't really have an opinion on usage of the `scope` parameter.

Reply:
I agree, the use of fixture_name and fixuture names almost repeating the exact same is not really DRY. 
In cases where we might shadow other variables one can change the name.

Regarding the scope - I will need to read up on it. But I am in favour of doing what is most efficient/ best practice in that instance.

Reply:
I agree as well! A bit unsure of how fixtures work but seems to me as good suggestions @seallard

Reply:
I think the practice of prefixing a fixture with `fixture_` comes from the instances where we directly import them. pytest actually picks up any fixtures annotated with the `fixture` decorator, so this is not necessary.

For our database fixtures, it requires some thought how to write and scope them. Attached a general summary about fixture scopes below generated by GPT:

<details>
  <summary><strong>Fixture Scopes</strong></summary>

  ### Use `function` scope when:
  - The fixture sets up state that should be isolated between tests.
  - The fixture performs actions like database transactions that need to be rolled back.
  - The setup and teardown actions are quick, and there's no performance concern.
  
  ### Use `class` scope when:
  - Multiple test methods within the same class can share the fixture without affecting each other.
  - You want to optimize performance by reducing setup/teardown for tests in the same class.
  
  ### Use `module` scope when:
  - The fixture data or state can safely be shared across multiple test functions in the same module.
  - Initialization of the fixture is resource-intensive, and you want to minimize that overhead.
  
  ### Use `session` scope when:
  - The fixture data or state is immutable or can be shared across all tests in the session.
  - The fixture initializes a resource that is expensive to create, like a database connection or a web server.
  
  ### Considerations:
  - **Resource Cleanup**: Be cautious about resource leaks when using broader scopes like `module` or `session`. Make sure to provide teardown logic if necessary.
  - **Test Isolation**: Broader scopes can lead to issues with test isolation. Make sure tests are still independent despite sharing fixtures.
  - **Debugging**: Tests that share fixtures at broader scopes can be harder to debug when they fail, as you may need to consider the shared state.
  - **Trade-off**: Changing the scope of a fixture is a trade-off between test isolation and performance. Choose the one that best suits your specific testing needs.
</details>